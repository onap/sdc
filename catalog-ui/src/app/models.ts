/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

/**
 * Created by ob0695 on 2/23/2017.
 */
import from = require("core-js/fn/array/from");
export * from './models/activity';
export * from './models/additional-information';
export * from './models/app-config';
export * from './models/validation-config';
export * from './models/plugins-config';
export * from './models/artifacts';
export * from './models/aschema-property';
export * from './models/schema-attribute';
export * from './models/attributes';
export * from './models/capability';
export * from './models/category';
export * from './models/comments';
export * from './models/components/component';
export * from './models/components/displayComponent';
export * from './models/components/resource';
export * from './models/components/service';
export * from './models/componentsInstances/componentInstance';
export * from './models/componentsInstances/resourceInstance';
export * from './models/componentsInstances/serviceInstance';
export * from './models/componentsInstances/serviceProxyInstance';
export * from './models/componentsInstances/serviceSubstitutionInstance';
export * from './models/graph/zones/group-instance';
export * from './models/graph/zones/policy-instance';
export * from './models/graph/zones/zone';
export * from './models/graph/zones/zone-instance';
export * from './models/csar-component';
//export * from './models/data-type-properties';
export * from './models/properties-inputs/property-be-model';
export * from './models/properties-inputs/property-fe-model';
export * from './models/properties-inputs/property-fe-map';
export * from './models/properties-inputs/derived-fe-property';
export * from './models/properties-inputs/property-declare-api-model';
export * from './models/properties-inputs/property-input-detail';
export * from './models/properties-inputs/input-fe-model';
export * from './models/properties-inputs/simple-flat-property';
export * from './models/data-types-map';
export * from './models/data-types';
export * from './models/distribution';
export * from './models/export-excel';
export * from './models/file-download';
export * from './models/graph/graph-links/common-base-link';
export * from './models/graph/graph-links/common-ci-link-base';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-link-base';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-simple-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-ucpe-host-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-ucpe-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-vl-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-vl-ucpe-link';
export * from './models/graph/graph-links/links-factory';
export * from './models/graph/graph-links/module-graph-links/module-ci-link-base';
export * from './models/graph/graph-links/module-graph-links/module-ci-vl-link';
export * from './models/graph/graphTooltip';
export * from './models/graph/assetPopoverObj';
export * from './models/graph/link-menu';
export * from './models/graph/match-relation';
export * from './models/graph/nodes/base-common-node';
export * from './models/graph/nodes/common-ci-node-base';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-base';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-cp';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-configuration';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service-proxy';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service-substitution';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-ucpe-cp';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-ucpe';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vf';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vfc';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vl';
export * from './models/graph/nodes/modules-graph-nodes/module-node-base';
export * from './models/graph/nodes/nodes-factory';
export * from './models/graph/point';
export * from './models/graph/connectRelationModel';
export * from './models/graph/relationship';
export * from './models/heat-parameters';
export * from './models/input-property-base';
export * from './models/inputs-and-properties';
export * from './models/inputs';
export * from './models/instance-inputs-properties-map';
export * from './models/left-panel';
export * from './models/member';
export * from './models/modules/base-module';
export * from './models/operation';
export * from './models/properties';
export * from './models/requirement';
export * from './models/server-error-response';
export * from './models/tab';
export * from './models/tooltip-data';
export * from './models/user';
export * from './models/validate';
export * from './models/component-metadata';
export * from './models/components-array';
export * from './models/modal';
export * from './models/button';
export * from './models/wizard-step';
export * from './models/radio-button';
export * from './models/filter-properties-assignment-data';
export * from './models/properties-inputs/input-be-model';
export * from './models/catalogSelector';
export * from './models/componentsInstances/fullComponentInstance';
export * from './models/catalogSelector';
export * from './models/componentsInstances/fullComponentInstance';
export * from './models/service-instance-properties-and-interfaces';
export * from './models/relationship-types';
export * from './models/tosca-presentation';
export * from './models/node-types';
export * from './models/capability-types';