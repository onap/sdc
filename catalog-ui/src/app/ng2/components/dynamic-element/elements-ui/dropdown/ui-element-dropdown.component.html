<select name='{{name}}' [(ngModel)]="value" #t (change)="onSave()"  [ngClass]="{'disabled':readonly}">
  <option *ngFor="let ddvalue of values" [value]="ddvalue.value">{{ddvalue.label}}</option>
</select>
