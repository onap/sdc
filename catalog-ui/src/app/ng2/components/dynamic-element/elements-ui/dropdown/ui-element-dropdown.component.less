@import '../../../../../../assets/styles/variables';

/deep/ ui-element-dropdown {

    select {
        text-indent: 6px;
        border: solid 1px @main_color_o;
        width: 100%;
    }

}
