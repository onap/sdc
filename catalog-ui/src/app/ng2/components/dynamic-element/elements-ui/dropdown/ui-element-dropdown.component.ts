/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

import { Component, EventEmitter, Output, Input } from '@angular/core'
import { BrowserModule } from '@angular/platform-browser'
import { UiElementBase, UiElementBaseInterface } from './../ui-element-base.component';

export class DropdownValue {
  value:string;
  label:string;

  constructor(value:string,label:string) {
    this.value = value;
    this.label = label;
  }
}

@Component({
  selector: 'ui-element-dropdown',
  templateUrl: './ui-element-dropdown.component.html',
  styleUrls: ['./ui-element-dropdown.component.less'],
})
export class UiElementDropDownComponent extends UiElementBase implements UiElementBaseInterface {
  @Input()
  values: DropdownValue[];

  constructor() {
    super();
  }

    onSave() {
        this.baseEmitter.emit(JSON.parse(this.value));
    }

}
