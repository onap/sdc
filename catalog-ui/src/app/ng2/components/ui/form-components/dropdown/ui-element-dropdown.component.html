<select name='{{name}}' [(ngModel)]="value" (change)="onSave()"  [ngClass]="{'disabled':readonly}" data-tests-id="SelectType">
  <option *ngFor="let ddvalue of values" [ngValue]="ddvalue.label != undefined ? ddvalue.value : ddvalue">{{ddvalue.label||ddvalue}}</option>
</select>
