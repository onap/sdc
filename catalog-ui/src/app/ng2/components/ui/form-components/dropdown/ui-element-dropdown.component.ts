/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

import {Component, EventEmitter, Output, Input} from '@angular/core'
import { UiElementBase, UiElementBaseInterface } from './../ui-element-base.component';

export class DropdownValue {
  value:any;
  label:string;
  hidden?: boolean;
  selected?: boolean;


  constructor(value:any,label:string, hidden: boolean = false, selected: boolean = false) {
    this.value = value;
    this.label = label;
    this.hidden = hidden;
    this.selected = selected;
    
    // this.hidden = hidden ? true : "hidden": "";
    // this.selected = selected ? true : "selected": "";
  }


}

@Component({
  selector: 'ui-element-dropdown',
  templateUrl: './ui-element-dropdown.component.html',
  styleUrls: ['./ui-element-dropdown.component.less'],
})
export class UiElementDropDownComponent extends UiElementBase implements UiElementBaseInterface  {
  
  @Input() values: DropdownValue[];

  constructor() {
    super();
  }
}
