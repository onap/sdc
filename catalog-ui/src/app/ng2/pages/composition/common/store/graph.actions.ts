export enum SelectedComponentType {
    COMPONENT_INSTANCE = "COMPONENT_INSTANCE",
    GROUP = "GROUP",
    POLICY = "POLICY",
    TOPOLOGY_TEMPLATE = "TOPOLOGY_TEMPLATE"
}

export class UpdateSelectedComponentAction {
    static readonly type = '[COMPOSITION] UpdateSelectedComponent';

        constructor(public payload: {uniqueId?: string, type?: string}) {
    }
}
    
export class SetSelectedComponentAction {
    static readonly type = '[COMPOSITION] SetSelectedComponent';

    constructor(public payload: {component?: any, type?: SelectedComponentType}) {
        }
}

export class OnSidebarOpenOrCloseAction {
    static readonly type = '[COMPOSITION] OnSidebarOpenOrCloseAction';

    constructor() {
    }
}

export class TogglePanelLoadingAction {
    static readonly type = '[COMPOSITION] TogglePanelLoading';
    constructor(public payload: { isLoading: boolean}) {
    }
}
