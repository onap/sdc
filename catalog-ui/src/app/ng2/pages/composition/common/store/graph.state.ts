import { Action, Selector, State, StateContext} from '@ngxs/store';
import {
    OnSidebarOpenOrCloseAction,
    SelectedComponentType,
    SetSelectedComponentAction,
    TogglePanelLoadingAction
} from "./graph.actions";
import { PolicyInstance, GroupInstance, Component as TopologyTemplate, ComponentInstance, LeftPaletteComponent, FullComponentInstance} from "app/models";
import { TopologyTemplateService } from "app/ng2/services/component-services/topology-template.service";
import { tap } from "rxjs/operators";
import { CompositionService } from "app/ng2/pages/composition/composition.service";
import {GroupsService} from "../../../../services/groups.service";
import {PoliciesService} from "../../../../services/policies.service";
import {WorkspaceService} from "../../../workspace/workspace.service";

export class CompositionStateModel {

    isViewOnly?: boolean;
    panelLoading?: boolean;
    selectedComponentType?: SelectedComponentType;
    selectedComponent?: PolicyInstance | GroupInstance | TopologyTemplate | ComponentInstance;
    withSidebar?: boolean;
}

@State<CompositionStateModel>({
    name: 'composition',
    defaults: {
         withSidebar: true
    }
})
export class GraphState {

    constructor(private topologyTemplateService: TopologyTemplateService,
                private compositionService: CompositionService,
         private policiesService:PoliciesService, private groupsService:GroupsService,
                 private workspaceService: WorkspaceService) {}

    @Action(SetSelectedComponentAction)
    setSelectedComponent({dispatch, getState, patchState}:StateContext<CompositionStateModel>, action: SetSelectedComponentAction) {
        
        const state:CompositionStateModel = getState();
        
        patchState({ panelLoading: true });

        if(action.payload.component instanceof ComponentInstance){
            let originComponent = this.compositionService.getOriginComponentById(action.payload.component.getComponentUid());
            if(!originComponent) {
                return this.topologyTemplateService.getFullComponent(action.payload.component.originType, action.payload.component.getComponentUid())
                .pipe(tap(resp => {
                    this.compositionService.addOriginComponent(resp);
                    this.compositionService.setSelectedComponentType(SelectedComponentType.COMPONENT_INSTANCE);
                    patchState({
                        selectedComponent: new FullComponentInstance(action.payload.component, resp), 
                        selectedComponentType: action.payload.type,
                        panelLoading: false
                    });
                    }, err => {
                        patchState({
                            panelLoading: false
                        })
                    }
                ));
            } else {
                patchState({
                    selectedComponent: new FullComponentInstance(action.payload.component, originComponent), 
                    selectedComponentType: action.payload.type,
                    panelLoading: false
                });
            }
        } else if (action.payload.component instanceof PolicyInstance) {
            let topologyTemplate = this.workspaceService.metadata;
            return this.policiesService.getSpecificPolicy(topologyTemplate.componentType, topologyTemplate.uniqueId, action.payload.component.uniqueId).pipe(tap(resp => 
                {
                    this.compositionService.updatePolicy(resp);
                    patchState({
                        selectedComponent: resp, 
                        selectedComponentType: action.payload.type,
                        panelLoading: false
                    })
                }, err => {
                    patchState({
                        panelLoading: false
                    })
                 }
            ));
        
        } else if (action.payload.component instanceof GroupInstance) {
            let topologyTemplate = this.workspaceService.metadata;
            return this.groupsService.getSpecificGroup(topologyTemplate.componentType, topologyTemplate.uniqueId, action.payload.component.uniqueId).pipe(tap(resp => {
                this.compositionService.updateGroup(resp);
                patchState({
                    selectedComponent: resp,
                    selectedComponentType: action.payload.type,
                    panelLoading: false
                });
            }, err => {
                patchState({
                    panelLoading: false
                })
             }
            ));
        } else { //TopologyTemplate
            patchState({
                selectedComponent: action.payload.component,
                selectedComponentType: action.payload.type,
                panelLoading: false
            })
        }
    }


    // @Action(UpdateSelectedComponentNameAction)
    // UpdateSelectedComponentNameAction({patchState}:StateContext<CompositionStateModel>, action: UpdateSelectedComponentNameAction) {

    //     switch(action.payload.type){
    //         case SelectedComponentType.COMPONENT_INSTANCE:
    //             this.store.dispatch(new UpdateComponentInstancesAction([action.payload.component]));
    //             break;
    //         case SelectedComponentType.POLICY:
    //             this.store.dispatch(new UpdatePolicyNameAction(action.payload.uniqueId, action.payload.newName));
    //             break;
    //         case SelectedComponentType.GROUP:
    //             this.store.dispatch(new UpdateGroupInstancesAction)

    //     }
    //     if(action.payload.type === SelectedComponentType.COMPONENT_INSTANCE){
            
    //     } 
    
    // }

    @Selector() 
    static getSelectedComponent(state:CompositionStateModel) {
        return state.selectedComponent;
    }

    @Selector() 
    static getSelectedComponentId(state:CompositionStateModel) {
        return state.selectedComponent.uniqueId;
    }

    @Selector() 
    static getSelectedComponentType(state:CompositionStateModel) {
        return state.selectedComponentType;
    }


    @Action(OnSidebarOpenOrCloseAction)
    onSidebarOpenOrCloseAction({getState, setState}:StateContext<CompositionStateModel>) {
        const state:CompositionStateModel = getState();

        setState({
            ...state,
            withSidebar: !state.withSidebar
        });
    }
    
    @Action(TogglePanelLoadingAction)
    TogglePanelLoading({patchState}:StateContext<CompositionStateModel>, action: TogglePanelLoadingAction) {

        patchState({
            panelLoading: action.payload.isLoading
        });
    }

    @Selector() static withSidebar(state):boolean {
        return state.withSidebar;
    }

}