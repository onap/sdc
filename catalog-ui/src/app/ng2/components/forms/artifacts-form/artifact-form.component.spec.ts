import {ArtifactFormComponent} from "./artifact-form.component";
import {async, ComponentFixture} from "@angular/core/testing";
import {ConfigureFn, configureTests} from "../../../../../jest/test-config.helper";
import {NO_ERRORS_SCHEMA} from "@angular/core";
import {TranslateModule} from "../../../shared/translator/translate.module";
import {CacheService} from "../../../services/cache.service";
import {TranslateService} from "../../../shared/translator/translate.service";
import {ArtifactModel} from "../../../../models/artifacts";
import {IDropDownOption} from "onap-ui-angular/dist/form-elements/dropdown/dropdown-models";
import {Observable, Subject} from "rxjs";
import {getValue} from "@ngxs/store";


describe('artifact form component', () => {

    let fixture: ComponentFixture<ArtifactFormComponent>;
    let cacheServiceMock: Partial<CacheService>;
    let onValidationChangeMock: Partial<Subject<boolean>>;

    let artifactModel = new ArtifactModel();


    beforeEach(
        async(() => {

            onValidationChangeMock = {
                next: jest.fn()
            }

            cacheServiceMock = {
                contains: jest.fn(),
                remove: jest.fn(),
                set: jest.fn(),
                get: jest.fn()
            }

            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [ArtifactFormComponent],
                    imports: [TranslateModule],
                    schemas: [NO_ERRORS_SCHEMA],
                    providers: [
                        {provide: CacheService, useValue: cacheServiceMock},
                        {provide: TranslateService, useValue: {}}
                    ],
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(ArtifactFormComponent);
            });
        })
    );


    it('should verify initArtifactTypes for DEPLOYMENT and ArtifactType = HEAT_ENV', () =>{

        cacheServiceMock.get.mockImplementation(() =>{
            return {
                artifacts: {
                    deployment:{
                        resourceInstanceDeploymentArtifacts: [{name: "Dummy Value Returned from ui api"}],
                    }
                }
            }
        });

        fixture.componentInstance.artifactType = 'DEPLOYMENT';
        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.artifact.artifactType = 'HEAT_ENV';

        fixture.componentInstance.initArtifactTypes();

        expect(fixture.componentInstance.selectedFileType).toEqual(undefined);

    });

    it('should verify initArtifactTypes for DEPLOYMENT and ComponentType = RESOURCE', () =>{

        cacheServiceMock.get.mockImplementation(() =>{
            return {
                artifacts: {
                    deployment:{
                        resourceDeploymentArtifacts: [{name: "Dummy Value Returned from ui api"}],
                    }
                }
            }
        });

        fixture.componentInstance.artifactType = 'DEPLOYMENT';
        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.componentType = 'RESOURCE';

        fixture.componentInstance.initArtifactTypes();

        expect(fixture.componentInstance.selectedFileType).toEqual(undefined);

    });

    it('should verify initArtifactTypes for DEPLOYMENT and NOT ComponentType = RESOURCE OR NOT ArtifactType = HEAT_ENV', () =>{

        cacheServiceMock.get.mockImplementation(() =>{
            return {
                artifacts: {
                    deployment:{
                        serviceDeploymentArtifacts: [{name: "Dummy Value Returned from ui api"}],
                    }
                }
            }
        });

        fixture.componentInstance.artifactType = 'DEPLOYMENT';
        fixture.componentInstance.artifact = artifactModel;
        // fixture.componentInstance.componentType = 'RESOURCE';

        fixture.componentInstance.initArtifactTypes();

        expect(fixture.componentInstance.selectedFileType).toEqual(undefined);

    });

    it('should verify initArtifactTypes for INFORMATION', () =>{

        cacheServiceMock.get.mockImplementation(() =>{
            return {
                artifacts: {
                    other: [{name: "Val1"}, {name: "ExpectedValToBeSelected"}, {name: "Val3"}]
                }
            }
        });

        fixture.componentInstance.artifactType = 'INFORMATIONAL';
        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.artifact.artifactType = 'ExpectedValToBeSelected';

        fixture.componentInstance.initArtifactTypes();

        expect(fixture.componentInstance.selectedFileType).toEqual({"label": "ExpectedValToBeSelected", "value": "ExpectedValToBeSelected"});

    });


    it('should match current snapshot of artifact form component', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('should verify onUploadFile -> file gets file name', () => {
        let file = {
            filename:'dummyFileName'
        }

        fixture.componentInstance.verifyTypeAndFileWereFilled = jest.fn();
        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.onUploadFile(file);

        expect(fixture.componentInstance.artifact.artifactName).toBe('dummyFileName');

        const spy1 = jest.spyOn(fixture.componentInstance,'verifyTypeAndFileWereFilled');
        expect(spy1).toHaveBeenCalled();
    });

    it('should verify onUploadFile -> file is null', () => {
        let file = null;
        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.onUploadFile(file);

        expect(fixture.componentInstance.artifact.artifactName).toBe(null);
    });

    it('should verify onTypeChange -> verifyTypeAndFileWereFilled is being called', () => {
        let selectedFileType:IDropDownOption;
        selectedFileType = {"label": "dummyLabel", "value": "dummyValue"};
        fixture.componentInstance.verifyTypeAndFileWereFilled = jest.fn();

        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.onTypeChange(selectedFileType);

        const spy1 = jest.spyOn(fixture.componentInstance,'verifyTypeAndFileWereFilled');
        expect(spy1).toHaveBeenCalled();
    });

    it('should verify onDescriptionChange -> verifyTypeAndFileWereFilled is being called', () => {
        fixture.componentInstance.verifyTypeAndFileWereFilled = jest.fn();
        fixture.componentInstance.onValidationChange.next = jest.fn(() => true);

        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.onDescriptionChange();

        const spy1 = jest.spyOn(fixture.componentInstance,'verifyTypeAndFileWereFilled');
        expect(spy1).toHaveBeenCalled();
    });

    it('should verify onLabelChange -> verifyTypeAndFileWereFilled is being called', () => {
        fixture.componentInstance.verifyTypeAndFileWereFilled = jest.fn();
        fixture.componentInstance.onValidationChange.next = jest.fn(() => true);

        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.onLabelChange(true);

        const spy1 = jest.spyOn(fixture.componentInstance,'verifyTypeAndFileWereFilled');
        expect(spy1).toHaveBeenCalled();
    });

    it('should verify verifyTypeAndFileWereFilled -> verify branch this.artifact.artifactType !== \'DEPLOYMENT\' ==>> onValidationChange.next(false)', () => {
        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.artifact.artifactType = 'NOT_DEPLOYMENT';
        fixture.componentInstance.artifact.mandatory = true;
        fixture.componentInstance.descriptionIsValid = false;

        let onValidationChangeResult;

        fixture.componentInstance.onValidationChange.subscribe((data) => {
            onValidationChangeResult = data;
            // console.log("Subscriber got data >>>>> "+ data);
        });

        fixture.componentInstance.verifyTypeAndFileWereFilled();

        expect(onValidationChangeResult).toBe(false);
    });

    it('should verify verifyTypeAndFileWereFilled -> verify branch this.artifact.artifactType !== \'DEPLOYMENT\' ==>> onValidationChange.next(true)', () => {
        fixture.componentInstance.artifact = artifactModel;
        fixture.componentInstance.artifact.artifactType = 'NOT_DEPLOYMENT';
        fixture.componentInstance.artifact.mandatory = true;
        fixture.componentInstance.artifact.artifactName = 'Something';
        fixture.componentInstance.descriptionIsValid = true;

        let onValidationChangeResult;

        fixture.componentInstance.onValidationChange.subscribe((data) => {
            onValidationChangeResult = data;
            // console.log("Subscriber got data >>>>> "+ data);
        });

        fixture.componentInstance.verifyTypeAndFileWereFilled();

        expect(onValidationChangeResult).toBe(true);
    });


});