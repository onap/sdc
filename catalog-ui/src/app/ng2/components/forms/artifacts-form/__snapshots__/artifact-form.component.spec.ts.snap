// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`artifact form component should match current snapshot of artifact form component 1`] = `
<artifact-form
  artifactTypesOptions={[Function Array]}
  cacheService={[Function Object]}
  initArtifactTypes={[Function Function]}
  onDescriptionChange={[Function Function]}
  onLabelChange={[Function Function]}
  onTypeChange={[Function Function]}
  onUploadFile={[Function Function]}
  onValidationChange={[Function Subject]}
  verifyTypeAndFileWereFilled={[Function Function]}
>
  <form
    class="artifact-form"
    name="artifactForm"
    novalidate=""
  >
    <onap-file-upload />
    <div
      class="artifact-form-container"
    >
      
      <div
        class="right-form-container"
      >
        <sdc-textarea
          label="Description"
          testid="description"
        />
        <sdc-validation>
          <sdc-required-validator />
          <sdc-regex-validator />
        </sdc-validation>
      </div>
    </div>
  </form>
</artifact-form>
`;
